speech/norm/__init__.py

```diff
@@ -231,42 +182,244 @@ def _email_handler(email: str) -> str:
-        "number": number_to_text,
-        "numbers": numbers_to_text,
-        # ^ Plural forms are lazy shortcuts for longer text
-        # TODO: amount/s
-        # TODO: currency/ies
-        # TODO: distance/s
-        "float": float_to_text,
-        "floats": floats_to_text,
-        "ordinal": number_to_ordinal,
-        "ordinals": numbers_to_ordinal,
-        "digits": digits_to_text,
-        "phone": digits_to_text,
-        "time": _time_handler,
-        "date": _date_handler,
-        "year": year_to_text,
-        "years": years_to_text,
-        "abbrev": _abbrev_handler,
-        "email": _email_handler,
-        "break": _break_handler,
-        "paragraph": lambda txt: f"<p>{txt}</p>",
-        "sentence": lambda txt: f"<s>{txt}</s>",
+class NormalizationHandler:  # Abstract base class
+    ...
+
+
+class DefaultNormalization(NormalizationHandler):
+    """
+    Class containing default text normalization functions
+    for Icelandic speech synthesis.
+    """
+
+    # TODO
+    # amount/s
+    # currency/ies
+    # distance/s
+
+    @classmethod
+    def number(cls, txt: str, **kwargs: str):
+        """Voicify a number."""
+        return number_to_text(txt, **kwargs)
+
+    @classmethod
+    def numbers(cls, txt: str, **kwargs: str):
+        """Voicify text containing multiple numbers."""
+        return numbers_to_text(txt, **kwargs)
+
+    @classmethod
+    def float(cls, txt: str, **kwargs: str):
+        """Voicify a float."""
+        return float_to_text(txt, **kwargs)
+
+    @classmethod
+    def floats(cls, txt: str, **kwargs: str):
+        """Voicify text containing multiple floats."""
+        return floats_to_text(txt, **kwargs)
+
+    @classmethod
+    def ordinal(cls, txt: str, **kwargs: str):
+        """Voicify an ordinal."""
+        return number_to_ordinal(txt, **kwargs)
+
+    @classmethod
+    def ordinals(cls, txt: str, **kwargs: str):
+        """Voicify text containing multiple ordinals."""
+        return numbers_to_ordinal(txt, **kwargs)
+
+    @classmethod
+    def digits(cls, txt: str):
+        """Spell out digits."""
+        return digits_to_text(txt)
+
+    @classmethod
+    def phone(cls, txt: str):
+        """Spell out digits."""
+        return cls.digits(txt)
+
+    @classmethod
+    def time(cls, txt: str):
+        """
+        Voicifies time of day, specified as 'HH:MM'.
+        E.g.
+            "11:34" -> "ellefu þrjátíu og fjögur",
+            "00:30" -> "tólf þrjátíu um nótt"
+        Note: doesn't check for incorrect data, caller should handle.
+        """
+        h, m = [int(i) for i in txt.split(":")]
+        suffix: Optional[str] = None
+        # Some times
+        if h == 0:
+            # Call 00:00 "tólf á miðnætti"
+            # and 00:xx "tólf ... um nótt"
+            h = 12
+            suffix = "á miðnætti" if m == 0 else "um nótt"
+        elif 0 < h <= 5:
+            suffix = "um nótt"
+        elif h == 12 and m == 0:
+            suffix = "um hádegi"
```

Comment:
"á hádegi" would be better and conform to "á miðnætti"

Reply:
Fixed!